crates/sui-framework/sources/id.move

```diff
@@ -59,6 +59,12 @@ module sui::id {
         version: u64
     }
 
+    /// An object ID representing an ID that has been transferred during this transaction.
+    /// It can be used as the parent ID target for transferring child objects.
+    struct TransferredID has copy, drop {
```

Comment:
Shouldn't this have `store` as well (in case a parent wants to remember its children)?

Reply:
Ah, I think I see why you don't do this (`transfer_to_id` API would make less sense if this had `store`).

I do think one useful feature of `ChildRef` is that it serves as proof that this object ID is a child object of a certain parent object ID. It would be nice to have a type that serves that purpose for folks that want. The use-case case I'm thinking of is something like `accessories: vector<ChildId>` as a field in the parent object. `vector<ID>` works for this too, but there's no integrity guarantee behind the ID's in this approach.

Reply:
I had that in an early version of this PR, where `transfer_to_child` also gave back a `ChildID<T>`, and we could do that again if we wanted.